generators/entity-server/templates/src/main/java/package/domain/Entity.java.ejs

```diff
@@ -283,15 +293,15 @@ public class <%= asEntity(entityClass) %> implements Serializable {
     @Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
                 <%_ }
             }
-        } else if (databaseType === 'mongodb' || databaseType === 'couchbase') {
-            if (databaseType === 'mongodb') { _%>
+        } else if ((databaseType === 'mongodb' || databaseType === 'couchbase')) {
+            if (databaseType === 'mongodb' && !otherEntityEmbedded) { _%>
```

Comment:
Does `otherEntityEmbedded` mean that the other entity is embedded? In that case, I prefer `embeddedOtherEntity` or `otherEntityIsEmbedded` (but I prefer the first one for consistency)

Reply:
Yes, this is exactly what it means, I followed the same pattern for `otherEntity` variables (`otherEntityRelationshipName`, `otherEntityRelationshipNamePlural`, ...). I think it is explicit enough, but if you don't like it, I can rename it. In that case I prefer `otherEntityIsEmbedded`